src/bin/doodle/format/tar.rs

```diff
@@ -37,14 +37,27 @@ pub fn main(module: &mut FormatModule, base: &BaseModule) -> FormatRef {
         ]),
     );
 
-    let cstr_arr =
-        |len: u16| -> Format { Format::Slice(Expr::U16(len), Box::new(tar_asciiz.call())) };
+    const MAGIC: &[u8; 6] = b"ustar\x00";
 
-    // A format for uninterpreted N-byte values
-    let cbytes = |len: u16| -> Format { repeat_count(Expr::U16(len), base.u8()) };
+    let nul_or_wsp = module.define_format(
+        "tar.padding-char",
+        Format::Byte(ByteSet::from([0x00, b' '])),
+    );
 
-    const MAGIC: &[u8; 6] = b"ustar\x00";
+    // A format for uninterpreted N-byte values
```

Comment:
Are these uninterpreted N-byte values? It appears to be an octal string :thinking:

Reply:
The comment is perhaps a bit out-of-date. I will do a bit of digging to re-confirm that only octal digits are allowed, rename the binding, and amend the comment.

Reply:
I have attempted to solve the issue, let me know if it is better now.